File: VisualStudio_IDE/MmgGameApiCs/MmgGameApiCs/src/net/middlemind/MmgGameApiCs/MmgCore/GenericEventHandler.cs
using System;

namespace net.middlemind.MmgGameApiCs.MmgCore
{
    /// <summary>
    /// An interface class for handling generic events. 
    /// Generic events are events that are not necessarily driven by user interaction.
    /// Created by Middlemind Games 08/01/2020
    /// </summary>
    public interface GenericEventHandler
    {
        /// <summary>
        /// Method for handling generic events.
        /// </summary>
        /// <param name="obj">A generic event message used to determine what to do in the actual implementation of this method.</param>
        public void HandleGenericEvent(GenericEventMessage obj);
    }
}

File: NetBeans_IDE/MmgGameApiJava/src/net/middlemind/MmgGameApiJava/MmgCore/GenericEventHandler.java
package net.middlemind.MmgGameApiJava.MmgCore;

/**
 * An interface class for handling generic events. 
 * Generic events are events that are not necessarily driven by user interaction. 
 * Created by Middlemind Games 08/01/2020
 *
 * @author Victor G. Brusca
 */
public interface GenericEventHandler {

    /**
     * Method for handling generic events.
     *
     * @param obj       A generic event message used to determine what to do in the actual implementation of this method.
     */
    public void HandleGenericEvent(GenericEventMessage obj);

}